src/app/components/media/ItemTitle.js

```diff
@@ -0,0 +1,291 @@
+            />
+          }
+        />
+        <ItemTitleOption
+          fieldName="pinned_media_id"
+          optionIcon={<PermMediaIcon />}
+          label={
+            <FormattedMessage
+              id="itemTitle.pinnedMediaId"
+              defaultMessage="Pinned Media ID"
+              description="Label for a menu item that selects Pinned Media ID as the title field for an item."
+            />
+          }
+        />
+        <ItemTitleOption
+          fieldName="claim_title"
+          optionIcon={<NoteAltIcon />}
+          disabled={!projectMedia.claim_description}
+          label={
+            <FormattedMessage
+              id="itemTitle.claimTitle"
+              defaultMessage="Claim Title"
+              description="Label for a menu item that selects Claim Title as the title field for an item."
+            />
+          }
+          helperText={
+            <FormattedMessage
+              id="itemTitle.claimTitleHelper"
+              defaultMessage="(Add a claim to enable)"
+              description="Helper text displayed under the Claim Title option for an item title when the item has no claim."
+            />
+          }
+        />
+        <ItemTitleOption
+          fieldName="fact_check_title"
+          optionIcon={<FactCheckIcon />}
+          disabled={!projectMedia.claim_description?.fact_check}
+          label={
+            <FormattedMessage
+              id="itemTitle.factCheckTitle"
+              defaultMessage="Fact-Check Title"
+              description="Label for a menu item that selects Fact-Check Title as the title field for an item."
+            />
+          }
+          helperText={
+            <FormattedMessage
+              id="itemTitle.factCheckTitleHelper"
+              defaultMessage="(Add a fact-check to enable)"
+              description="Helper text displayed under the Fact-Check Title option for an item title when the item has no fact-check."
+            />
+          }
+        />
+      </Menu>
+    </div>
+  );
+};
+
+ItemTitle.propTypes = {
+  projectMedia: PropTypes.shape({
+    id: PropTypes.string.isRequired,
+    permissions: PropTypes.string.isRequired, // { key => value } (e.g., { 'update ProjectMedia' => true }), as a JSON string
+    archived: PropTypes.number.isRequired,
+    title: PropTypes.string.isRequired,
+    title_field: PropTypes.oneOf(['custom_title', 'pinned_media_id', 'claim_title', 'fact_check_title']),
+    custom_title: PropTypes.string,
+    media_slug: PropTypes.string,
+    claim_description: PropTypes.shape({
+      description: PropTypes.string.isRequired,
+      fact_check: PropTypes.shape({
+        title: PropTypes.string.isRequired,
+      }),
+    }),
+  }).isRequired,
+};
+
+// For unit tests
+// eslint-disable-next-line import/no-unused-modules
+export { ItemTitle };
+
+export default createFragmentContainer(ItemTitle, {
```

Comment:
Ideally I would like to see this component as a `QueryRenderer` passed a `dbid` for the `project_media` query, like we do in `MediaTags`:

https://github.com/meedan/check-web/blob/develop/src/app/components/media/MediaTags.js

Especially since we don't really need an `ItemTitle_projectMedia` fragment, right?

In `MediaTags` we pass in `projectMediaId` and then every piece derives from the query based on that id. So you can see in `MediaActionsBar` right now that the components are called like:

```jsx
          <ItemTitle projectMedia={this.props.media} />
          <MediaTags projectMediaId={this.props.media?.dbid} />
```

What I'd like to see is

```jsx
          <ItemTitle projectMediaId={this.props.media?.dbid} />
          <MediaTags projectMediaId={this.props.media?.dbid} />
```

Basically it makes it clear that all we really need for `ItemTitle` to work is the dbid, and that it's all self contained from a query, rather than receiving a huge projectmedia object.

Reply:
I shared this with @dariusk privately, but just to document it here as well:

There is an issue that happens on Relay when there are two `QueryRenderer`'s for the same route at the same level.

For example, for the suggestion below:

```jsx
  <ItemTitle projectMediaId={this.props.media?.dbid} />
  <MediaTags projectMediaId={this.props.media?.dbid} />
```

So, if `<ItemTitle />` has a `<QueryRenderer />` that asks for `project_media { title }` and `<MediaTags />` has a `<QueryRenderer />` that asks for `project_media { tags }`, this is what happens:

- If `<ItemTitle />` comes first, then `project_media.title` will be present but `project_media.tags` will be null, so `<MediaTags />` crashes
- If `<MediaTags />` comes first, then `project_media.tags` will be present but `project_media.title` will be null, so `<ItemTitle />` crashes

So, the `project_media` returned by one of the `QueryRenderer`'s fully replaces (in the Relay store) the one returned by the other `QueryRenderer` instead of merging them... it's as if Relay "forces" you to avoid more than one `<QueryRenderer />` for the same page.

I worked around that by differentiating the `variables` (the `QueryRenderer` `prop`) names between these components.

It also made me realize that it's important for us to document in our frontend good practices document when to use a fragment container or a query renderer, and other details like this issue I just described.